Refactor: Optimize create_test_file in benchmark.py

mirusu400 opened the pull request:
The original function used a while loop to write one random byte at a time, so I use `os.urandom(size)` to efficiently generate the full size bytes of random data in memory.

If the requested size is too large to fit in available system RAM (e.g., 50GB file on a 16GB machine), a MemoryError is caught. In this scenario, the function falls back to the original, memory-efficient byte-by-byte write logic.

lxp commented:
Thanks for your pull request!

Ideally, I would like to get rid of the old behavior at all, since it is very inefficient.
However, reading a huge block of random data at once before writing it is also not very efficient memory wise.

Can you maybe come up with a middle way like reading and writing blocks of 64k?
At least 64k is also used when hashing files without mmap support: https://github.com/cfv-project/cfv/blob/79587c1ff05169ad67342e6fbc6a3c59dfec251a/lib/cfv/hash.py#L40

Please also fix the formatting error (see https://github.com/cfv-project/cfv/actions/runs/18862085679/job/54029609083?pr=84):
```
Run flake8 --exclude=build,venv --ignore= --max-line-length=200 --max-complexity=75 --show-source --statistics .
./test/benchmark.py:69:1: W293 blank line contains whitespace
    
^
1     W293 blank line contains whitespace
```

mirusu400 committed:
Refractor: Optimize create_test_file with memory-safe fallback

mirusu400 commented:
Hello @lxp, I agree with your suggestion

I changed the code to write files split by chunk, not the whole data.

Also, I fixed the flake8 formatting error

Thank you for quick review!

lxp commented on test/benchmark.py:
```diff
@@ -62,15 +62,32 @@ def human_int(value):
 
 
-def create_test_file(path, max_size, verbose=False):
+def create_test_file(path, max_size, verbose=False, chunk_size=65535):
```
We should use a power of 2 for the chunk_size.

```suggestion
def create_test_file(path, max_size, verbose=False, chunk_size=65536):
```

lxp commented on test/benchmark.py:
```diff
@@ -62,15 +62,32 @@ def human_int(value):
+    
+    size = random.randint(1, max_size)
+    if verbose:
+        print("creating", path, "size", size, "chunk_size", chunk_size)
```
This is duplicate code we can remove. Also printing the chunk_size is not necessary as it is only internal information not relevant to the user. 

```suggestion
```

lxp commented on test/benchmark.py:
```diff
@@ -62,15 +62,32 @@ def human_int(value):
-def create_test_file(path, max_size, verbose=False):
+def create_test_file(path, max_size, verbose=False, chunk_size=65535):
+    if max_size < 1:
+        raise ValueError("max_size must be >= 1")
```
The codebase uses single ticks for strings whenever it is possible without complicating things.

```suggestion
        raise ValueError('max_size must be >= 1')
```

lxp commented on test/benchmark.py:
```diff
@@ -62,15 +62,32 @@ def human_int(value):
+    if max_size < 1:
+        raise ValueError("max_size must be >= 1")
+    if chunk_size < 1:
+        raise ValueError("chunk_size must be >= 1")
```
Again single ticks for strings.

```suggestion
        raise ValueError('chunk_size must be >= 1')
```

lxp commented on test/benchmark.py:
```diff
@@ -62,15 +62,32 @@ def human_int(value):
+
+    written = 0
+    try:
+        with open(path, "wb") as f:
```
Again single ticks for strings.

```suggestion
        with open(path, 'wb') as f:
```

lxp commented on test/benchmark.py:
```diff
@@ -62,15 +62,32 @@ def human_int(value):
+        return written
+    except OSError as e:
+        # Disk full, permission issues, etc. Propagate with context.
+        raise OSError(f"Failed to write {path} after {written} bytes: {e}") from e
```
Again single ticks for strings.

```suggestion
        raise OSError(f'Failed to write {path} after {written} bytes: {e}') from e
```

lxp commented on test/benchmark.py:
```diff
@@ -62,15 +62,32 @@ def human_int(value):
+                written += n
+        return written
+    except OSError as e:
+        # Disk full, permission issues, etc. Propagate with context.
```
I don't think this comment is necessary.

```suggestion
```

lxp commented on test/benchmark.py:
```diff
@@ -62,15 +62,32 @@ def human_int(value):
-        while size:
-            f.write(b'%c' % random.randint(0, 255))
-            size -= 1
+    
```
There is still a flake8 error on this line.

```
Run flake8 --exclude=build,venv --ignore= --max-line-length=200 --max-complexity=75 --show-source --statistics .
./test/benchmark.py:73:1: W293 blank line contains whitespace
    
^
1     W293 blank line contains whitespace
Error: Process completed with exit code 1.
```

See: https://github.com/cfv-project/cfv/actions/runs/18927372363/job/54136536126?pr=84

```suggestion

```

lxp reviewed (changes requested):
Thanks for updating your pull request accordingly.
I made a few more suggestions with concrete changes.
We should be able to merge it with this changes (at least when there are not additonal flake8 formatting errors).

mirusu400 committed:
Update test/benchmark.py

Co-authored-by: Lisa Gnedt <lisa@gnedt.at>

mirusu400 committed:
Update test/benchmark.py

Co-authored-by: Lisa Gnedt <lisa@gnedt.at>

mirusu400 committed:
Update test/benchmark.py

Co-authored-by: Lisa Gnedt <lisa@gnedt.at>

mirusu400 committed:
Update test/benchmark.py

Co-authored-by: Lisa Gnedt <lisa@gnedt.at>

lxp committed:
Update test/benchmark.py

lxp committed:
Apply suggestions from code review
